frontend/src/api.js

```diff
@@ -13,7 +28,14 @@ class API {
         const params = new URLSearchParams();
         params.set('all', all);
         params.set('deleted_only', deletedOnly);
-        return fetch(`${this.base}/schemas?${params.toString()}`);
+        const response = await fetch(`${this.base}/schemas?${params.toString()}`);
+        try {
+            this._is_response_ok(response);
+            return response.json()
+        }
+        catch (e) {
+            // TODO: Do something else
+        }
```

Comment:
1. I'm okay with returning json directly from function, but why is it just here?🤔
2. I'm not sure that we should catch exceptions at API functions, or at least our default behavior should be to reraise them so that in views we could decide how to react on specific error

Reply:
> but why is it just here?

I did not get through all the views and code, I only got around to change this one. If you like the general idea, that it great to hear and I will continue with refactoring the other methods.

> I'm not sure that we should catch exceptions at API functions

I'm afraid that a lot of exception handling in the views would look very similar. So I was trying to centralize exception handling. but we could extend it by passing custom error handler functions to the API method calls :thinking:

Reply:
> passing custom error handler functions
This could possibly work, I guess. The idea is when getting exception, in most cases (if not all) user should see that something went wrong and preferably what exactly which is something that has to be defined in views. If vuejs functions keep correct `this` context when passed to other (non-vue) functions then great, we can try passing exception handlers from views to API calls

Reply:
Error and success messages are now pushed onto a queue from which they are displayed in the UI.